datahub-web-react/src/app/ingest/source/IngestionSourceList.tsx

```diff
@@ -294,6 +313,7 @@ export const IngestionSourceList = () => {
                             (recipeBuilderState.config?.executorId as string)) ||
                         DEFAULT_EXECUTOR_ID,
                     debugMode: recipeBuilderState.config?.debugMode || false,
+                    extraArgs: recipeBuilderState.config?.extraArgs || [],
```

Comment:
can you do the formatting of `extraArgs` here and pass into `createOrUpdateIngestionSource` properly formatted instead of having to do the whole shebang where you have to spread the whole object etc.?

so you could just do `extraArgs: sanitizeExtraArgs(recipeBuilderState.config?.extraArgs || []),`